astropy/cosmology/parameter/_core.py

```diff
@@ -143,14 +143,13 @@ def __post_init__(self) -> None:
         # attribute name on container cosmology class.
         # really set in __set_name__, but if Parameter is not init'ed as a
         # descriptor this ensures that the attributes exist.
-        object.__setattr__(self, "name", None)
-        object.__setattr__(self, "_attr_name", None)
+        self.__set_name__(None, None)
 
     def __set_name__(self, cosmo_cls: type, name: str) -> None:
         # attribute name on container cosmology class
         self._attr_name: str
         object.__setattr__(self, "name", name)
-        object.__setattr__(self, "_attr_name", "_" + name)
+        object.__setattr__(self, "_attr_name", "_" + (name or ""))
```

Comment:
I don't see why is this change in code is being made. According to the type hint `name` is `str` and the empty string is the only falsy string.

Reply:
Let me fix that. I see that the ``__set_name__`` constructor is being called with ``name=None`` in one location.

Reply:
Done.